### ISSUE TITLE
_requests.get() times out

### ISSUE THREAD

**Issue by beglitis** [ranaroussi/yfinance#445]
Unfortunately I started getting random timeouts in the following call in base.py

_requests.get(url=url, params=params, proxies=proxy)

I added some print statements to debug

yfinance: Downloading data for ticker TKA.DE
yfinance: https://query1.finance.yahoo.com/v8/finance/chart/TKA.DE

and opening the URL in firefox seems to be fine. 

I think this issue is coming from Yahoo's side, am I being throttled?

**Comment by glebsam** [comment]
I have this issue as well

**Comment by beglitis** [comment]
@glebsam That is not good....

**Comment by uhoenig** [comment]
same for me :/

**Comment by beglitis** [comment]
@uhoenig I don't think it is ticker dependent, it happens randomly and if I pick the URL and open it in the browser it works. Did you try to download the entire history of a ticker or just a single value/date ?

**Comment by uhoenig** [comment]
you may be right, I get this time out at random tickers. I have a list of 50 and go back 5 yrs. That means for "younger" stocks it goes up to 5 yrs, maybe less.

**Comment by uhoenig** [comment]
to be fair, i placed _time.sleep(0.5) in multi.py in the for loop and it works (for now). I will observe how that works out

**Comment by beglitis** [comment]
@uhoenig It doesn't work for me even with the sleep() thrown in. Real bummer....

**Comment by SebastianGraeber** [comment]
Same for me, but the normal pandas_datareader doesn't work as well 
Seems something with Yahoo itself

**Comment by beglitis** [comment]
OK so I think Yahoo had enough of us and they decided to block scrappers. Fine.

**Comment by uhoenig** [comment]
what alternatives do we have though?

**Comment by beglitis** [comment]
OK I have a fix in place... did a few runs and worked for me but I haven't tested it under heavy loads to check how Yahoo will respond. Let me commit it back to my fork and I'll if I can do a merge request etc

**Comment by beglitis** [comment]
Try this

https://github.com/ranaroussi/yfinance/pull/449/commits/6a8c637af758ff3de53f2ca3dafb6bf10f1902b9

**Comment by uhoenig** [comment]
i set it up as well and it also works for me so far. i will still keep some lag for the moment, but may remove it if proves well. thank you, I appreciate it!

**Comment by bardvir** [comment]
This only seem to work for me if I run it single threaded `yf.download(threads=False)`

Otherwise, window can't keep up with the pace and it closes unexpectedly, then I get this error message:
selenium.common.exceptions.NoSuchWindowException: Message: no such window: target window already closed from unknown error: web view not found

Can anyone suggest a fix? working with a single thread slows down my work significantly

**Comment by jscmal** [comment]
Hi, it does not work for me too. I use `( threads = True )`

Can anyone with the right skill, try to solve this problem?
